onchain/cairo/launchpad/src/launchpad/locker/lock_manager.cairo

```diff
@@ -297,27 +272,27 @@ pub mod LockManager {
         }
 
         fn user_locks_length(self: @ContractState, user: ContractAddress) -> u32 {
-            let user_locks: Vec<felt252> = self.user_locks.read(user);
-            user_locks.len
+            let user_locks_length = self.user_locks_list.entry(user).len();
+
+            user_locks_length.try_into().unwrap()
         }
 
         fn user_lock_at(
             self: @ContractState, user: ContractAddress, index: u32
         ) -> ContractAddress {
-            let user_locks: Vec<felt252> = self.user_locks.read(user);
-            user_locks[index]
+            self.user_locks_list.entry(user).at(index.into()).read()
         }
 
         fn token_locks_length(self: @ContractState, token: ContractAddress) -> u32 {
-            let list: Vec<felt252> = self.token_locks.read(token);
-            list.len()
+            let token_locks_length = self.token_locks_list.entry(token).len();
```

Comment:
Never used, so let's do it please add a counter index with a Map or a way to get len of Map